src/ucp/core/ucp_request.inl

```diff
@@ -417,6 +417,21 @@ ucp_request_send_state_advance_comp(ucp_request_t *req, ucs_status_t status)
     }
 }
 
+static UCS_F_ALWAYS_INLINE int
+ucp_request_send_state_check_comp(ucp_request_t *req)
+{
+    ucs_assertv(req->send.state.dt.offset <= req->send.length,
+                "req=%p offset=%zu length=%zu", req, req->send.state.dt.offset,
+                req->send.length);
+
+    if ((req->send.state.dt.offset != req->send.length) &&
+        ucs_likely(req->send.state.uct_comp.status == UCS_OK)) {
+        return 0;
+    }
+
+    return 1;
+}
```

Comment:
return (req->send.state.dt.offset == req->send.length) ||
        (req->send.state.uct_comp.status != UCS_OK);
is this fast path?

Reply:
done